lib/pure/streams.nim

```diff
@@ -437,6 +438,71 @@ when not defined(js):
     else:
       raise newEIO("cannot open file")
 
+  type
+    MemMapFileStream* = ref MemMapFileStreamObj ## a stream that encapsulates a `MemFile`
```

Comment:
This should be in the ``memfiles`` module.

Reply:
And StringStream in the strutils?

Reply:
No because ``string`` is in ``system``.

Reply:
So, move to memfiles?

Reply:
Yeah, please.